aten/src/ATen/native/Pow.cpp

```diff
@@ -81,6 +81,60 @@ Tensor pow(Scalar base, const Tensor& exp) {
   return native::pow_out(result, base, exp);
 }
 
+Tensor& float_power_out(Tensor& result, const Tensor& base, const Tensor& exp) {
+  auto dtype = (at::isComplexType(base.scalar_type()) || at::isComplexType(exp.scalar_type())) ?
+                at::kComplexDouble : at::kDouble;
+  TORCH_CHECK(result.scalar_type() == dtype,
+              "output type ", result.scalar_type(), "is not the desired output type ", dtype);
+
+  return at::pow_out(result, base.to(dtype), exp.to(dtype));
+}
+
+Tensor& float_power_out(Tensor& result, const Tensor& base, Scalar exp) {
+  auto dtype = (at::isComplexType(base.scalar_type()) || exp.isComplex()) ? at::kComplexDouble : at::kDouble;
+  TORCH_CHECK(result.scalar_type() == dtype,
+              "output type ", result.scalar_type(), "is not the desired output type ", dtype);
+
+  return at::pow_out(result, base.to(dtype), exp);
+}
+
+Tensor& float_power_out(Tensor& result, Scalar base, const Tensor& exp) {
+  auto dtype = (base.isComplex() || at::isComplexType(exp.scalar_type())) ? at::kComplexDouble : at::kDouble;
+  TORCH_CHECK(result.scalar_type() == dtype,
+              "output type ", result.scalar_type(), "is not the desired output type ", dtype);
+  return at::pow_out(result, c10::scalar_to_tensor(base, exp.device()), exp.to(dtype));
+}
+
+Tensor float_power(const Tensor& base, const Tensor& exp) {
+  auto dtype = (at::isComplexType(base.scalar_type()) || at::isComplexType(exp.scalar_type())) ? at::kComplexDouble : at::kDouble;
+  Tensor result = at::empty({0}, base.options().dtype(dtype));
+  return native::float_power_out(result, base, exp);
+}
+
+Tensor float_power(const Tensor& base, Scalar exp) {
+  auto dtype = (at::isComplexType(base.scalar_type()) || exp.isComplex()) ? at::kComplexDouble : at::kDouble;
+  Tensor result = at::empty_like(base, base.options().dtype(dtype), MemoryFormat::Preserve);
+  return native::float_power_out(result, base, exp);
+}
+
+Tensor float_power(Scalar base, const Tensor& exp) {
+  auto dtype = (base.isComplex() || at::isComplexType(exp.scalar_type())) ? at::kComplexDouble : at::kDouble;
+  Tensor result = at::empty_like(exp, exp.options().dtype(dtype), MemoryFormat::Preserve);
+  return native::float_power_out(result, base, exp);
+}
+
+Tensor& float_power_(Tensor& base, const Tensor& exp) {
+  auto dtype = (at::isComplexType(base.scalar_type()) || at::isComplexType(exp.scalar_type())) ? at::kComplexDouble : at::kDouble;
+  base = base.to(dtype);
```

Comment:
The inplace variant has to already have the correct dtype since inplace operations change values but not dtypes or sizes. I think this variant (and the next one) should just verify the dtype of base is correct, cast exp to the appropriate dtype (if necessary), and then call `pow_`.

Reply:
Addressed, Thanks for the explanation！